json/targeting.yaml

```
@@ -0,0 +1,412 @@
+        the associated value from the "variants" object. Be sure that the returned
+        string is present as a key in the variants!.
+      type: string
+  varRule:
+    title: Var Operation
+    description: Retrieve data from the provided data object.
+    type: object
+    additionalProperties: false
+    properties:
+      var:
+        anyOf:
+        - type: string
+          description: flagd automatically injects "$flagd.timestamp" (unix epoch)
+            and "$flagd.flagKey" (the key of the flag in evaluation) into the context.
+          pattern: "^\\$flagd\\.((timestamp)|(flagKey))$"
+        - not:
+            $comment: this is a negated (not) match of "$flagd.{some-key}", which is faster and more compatible that a negative lookahead regex
+            type: string
+            description: flagd automatically injects "$flagd.timestamp" (unix epoch) and "$flagd.flagKey"
+              (the key of the flag in evaluation) into the context.
+            pattern: "^\\$flagd\\..*$"
+        - type: array
+          $comment: this is to support the form of var with a default... there seems to be a bug here, where ajv gives a warning (not an error) because maxItems doesn't equal the number of entries in items, though this is valid in this case
+          minItems: 1
+          items:
+            - type: string
+          additionalItems: 
+            anyOf:
+              - type:
+                  'null'
+              - type:
+                  boolean
+              - type: 
+                  string
+              - type:
+                  number
+  missingRule:
+    title: Missing Operation
+    description: Takes an array of data keys to search for (same format as var). Returns
+      an array of any keys that are missing from the data object, or an empty array.
+    type: object
+    additionalProperties: false
+    properties:
+      missing:
+        type: array
+        items:
+          type: string
+  missingSomeRule:
+    title: Missing-Some Operation
+    description: Takes a minimum number of data keys that are required, and an array
+      of keys to search for (same format as var or missing). Returns an empty array
+      if the minimum is met, or an array of the missing keys otherwise.
+    type: object
+    additionalProperties: false
+    properties:
+      missing_some:
+        minItems: 2
+        maxItems: 2
+        type: array
+        items:
+        - type: number
+        - type: array
+          items:
+            type: string
+  ifRule:
+    type: object
+    additionalProperties: false
+    properties:
+      if:
+        title: If Operator
+        description: 'The if statement takes 3 arguments: a condition (if), what to
+          do if its true (then), and what to do if its false (else). Note that the
+          form accepting more than 3 arguments (if/else) is not supported in flagd;
+          use nesting instead.'
+        type: array
+        minItems: 3
```

```suggestion
        minItems: 2
```

The else block is optional and works fine if omitted.